hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp

```diff
@@ -316,13 +320,81 @@ namespace hpx { namespace threads { namespace detail
     }
 
     template <typename Scheduler>
-    void scheduled_thread_pool<Scheduler>::resume(error_code& ec)
+    template <typename F>
+    void scheduled_thread_pool<Scheduler>::suspend_func(F&& callback, error_code& ec)
     {
-        if (!(mode_ & threads::policies::enable_elasticity))
+        for (std::size_t k = 0;
+             sched_->Scheduler::get_thread_count() >
+                 get_background_thread_count();
+             ++k)
+        {
+            util::detail::yield_k(k, "scheduled_thread_pool::suspend_func");
+        }
+
+        for (std::size_t i = 0; i != threads_.size(); ++i)
+        {
+            suspend_processing_unit_internal(i, ec);
+        }
+
+        callback();
+    }
+
+    template <typename Scheduler>
+    template <typename F>
+    void scheduled_thread_pool<Scheduler>::suspend_internal(F&& callback, error_code& ec)
+    {
+        if (threads::get_self_ptr() && hpx::this_thread::get_pool() == this)
         {
+            HPX_THROWS_IF(ec, bad_parameter,
+                "scheduled_thread_pool<Scheduler>::suspend",
+                "cannot suspend a pool from itself");
             return;
         }
 
+        auto suspend_func_wrapper =
+            [this, callback{std::move(callback)}, ec]() mutable
```

Comment:
This is C++14 only. We try to stay compatible with C++11. Please use the (new) `HPX_CAPTURE_FORWARD` macro instead of the direct initialization of the `callback` capture variable (see https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/config/lambda_capture.hpp).

Reply:
Yep, I'm aware. Again needs tidying up. Sorry, should've been more explicit about what's still to do.

Reply:
I C++11 we just copy into the capture :/ The only alternative is to use `bind`/`deferred_call` instead of the lambda.

Reply:
I had problems with the `packaged_task` as it's non-copyable (I need the move). But I think I can do without the `packaged_task` and just do an `hpx::async` for that case (IMO neither more complicated nor simpler). Otherwise it's like you say, `bind`/`deferred_call`.